contracts/swap/swap.go

```diff
@@ -152,6 +180,34 @@ func (s simpleContract) PaidOut(opts *bind.CallOpts, addr common.Address) (*big.
 	return s.instance.PaidOut(opts, addr)
 }
 
+// TotalDeposit iterates over all Deposit events and returns the total amount ever deposited
+func (s simpleContract) TotalDeposit() (totalDeposit *big.Int, err error) {
+	depositIterator, err := s.instance.FilterDeposit(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	for depositIterator.Next() {
+		totalDeposit = totalDeposit.Add(totalDeposit, depositIterator.Event.Amount)
+	}
+
+	return totalDeposit, depositIterator.Error()
```

Comment:
check the value of `err = depositIterator.Error()` first.

if `nil`, return `totalDeposit, nil`. 

else, return `&big.Int{}, err`.

Reply:
ok!